UndertaleModLib/Models/UndertaleTexturePageItem.cs

```diff
@@ -76,5 +86,99 @@ public override string ToString()
         {
             return "(" + GetType().Name + ")";
         }
+
+        public void ExportAsPNG(UndertaleData data, string FullPath, string imageName, bool newTarget)
+        {
+            int exportWidth = BoundingWidth; // sprite.Width
+            int exportHeight = BoundingHeight; // sprite.Height
+            if (newTarget)
+                renderTarget = new WriteableBitmap(exportWidth, exportHeight, 96, 96, PixelFormats.Pbgra32, null);
+
+            // Clear the render target.
+            if (pixelBuffer == null)
+            {
+                foreach (var texPageItem in data.TexturePageItems)
+                {
+                    if ((int)texPageItem.BoundingWidth > maxSpriteWidth)
+                        maxSpriteWidth = texPageItem.BoundingWidth;
+                    if ((int)texPageItem.BoundingHeight > maxSpriteHeight)
+                        maxSpriteHeight = texPageItem.BoundingHeight;
+                }
```

Comment:
What if somebody adds a new, big TexturePageItem after the buffer was already allocated?

Reply:
Good point. This is why we'll be making a TextureExporter class which can be instanced and thrown out.